### PR TITLE
Use new `dub add` command instead of copying json/sdl

### PR DESCRIPTION
**Pull request by Flying-Toast**
**This PR shouldn't be merged until a new dub version is released with dlang/dub#1587.**

dlang/dub#1587 creates an `add` command that adds a dependency, so instead of copy/pasting `"vibe-d": "~>0.8.4"` you just have to run `dub add vibe-d`.

### CODE REVIEW

**Comment by thewilsonator** [comment]
Please ping me when that happens, I'm likely to forget.

**Comment by MartinNowak** [comment]
Mmh dub add doesn't support specific versions (e.g. see [A](http://code.dlang.org/packages/vibe-d/0.8.4) and [B](http://code.dlang.org/packages/vibe-d/0.7.30)).
Most of the time only the latest is needed, but would be nice if that was supported, e.g. as `dub add vibe-d='~>0.8.4' dpq2='~>1.0.5'`. 
Shouldn't be too hard to still add version support to dub's add command before we release 2.084.0.

**Comment by MartinNowak** [comment]
See https://github.com/dlang/dub/pull/1626 for version-spec support in `dub add` and `dub init`.

**Comment by Flying-Toast** [comment]
@thewilsonator 2.084.0 was released.
Do you think it would be better to wait a while longer, to give people time to update?

**Comment by wilzbach** [comment]
Let's wait at least until LDC's next release with 2.084.

**Comment by Flying-Toast** [comment]
@wilzbach LDC 1.14.0 was released yesterday :tada:

**Comment by Flying-Toast** [comment]
Oops. Why is the "Close and comment" button right next to the "Comment" button?! :laughing:

File: views/view_package.dt [hunk 1]
```diff
@@ -64,20 +64,13 @@ block body
...
-				.clipboard-pkg-title dub.json
-				input(id="package-clipboard-json",value='"#{packageName}": "#{expr}"')
-				button(class="btn-clipboard",data-clipboard-target="#package-clipboard-json")
-					img(src="/images/clippy.svg", width=13, alt="Copy to clipboard")
-
-			.clipboard-pkg-row
-				.clipboard-pkg-title dub.sdl
-				input(id="package-clipboard-sdl",value='dependency "#{packageName}" version="#{expr}"')
-				button(class="btn-clipboard",data-clipboard-target="#package-clipboard-sdl")
+				input(id="package-clipboard",value='dub add #{packageName}')
```
Comment (wilzbach) [inline views/view_package.dt]: Shouldn't we include the version here?

**Comment by wilzbach** [comment]
> @wilzbach LDC 1.14.0 was released yesterday

Hmm, but it takes quite some time until at least the majority of users (and distro) have upgraded to it, no?
The upcoming disco release of Ubuntu (19.04) will most likely ship with LDC 1.12 (and Dub 1.12):

https://packages.ubuntu.com/disco/ldc
https://packages.ubuntu.com/disco/dub

> Do you think it would be better to wait a while longer, to give people time to update?

Maybe as a compromise we can show both instructions for this intermediate period?

**Comment by ZombineDev** [comment]
As there are plenty of ways to obtain the latest version of dub (install.sh, snapd, tar.gz download, etc.) I see no reason not to show instructions for the latest and greatest version.
If we really want, we could display an ℹ️ info tooltip reminding users that they need a recent version of dub.

File: views/view_package.dt (hunk 1 above)
Comment (Flying-Toast) [reply views/view_package.dt]: `dub add` will query the registry for the latest version. `dub add packageName=1.2.3` would do the same thing as just `dub add packageName`

Comment (wilzbach) [reply views/view_package.dt]: Yes, but by supplying the exact version we can make sure the user gets an error if there's an issue with the registry mirror he's using. Also currently we advertise the current version:

image

I had a quick look around. Cargo does it the same way, but PIP and NPM also don't use the exact version. Other opinions?

**Comment by wilzbach** [comment]
CC @s-ludwig and @MartinNowak. What do you think about this? Should we wait longer or release this now?

**Comment by wilzbach** [comment]
No one objected, so let's do this with the next release of the registry.

**Comment by Flying-Toast** [comment]
So should we also merge dlang/dub-docs#13 for consistency?

**Comment by 0xEAB** [comment]
This should really have been `additionally` instead of `instead` :roll_eyes:

**Pull request merged**